skbio/alignment/tests/test_tabular_msa.py

```diff
@@ -562,7 +630,7 @@ def test_reassign_index_with_mapping_callable(self):
         seqs = [DNA("A"), DNA("C"), DNA("G")]
 
         msa = TabularMSA(seqs, index=[0, 1, 2])
-        msa.reassign_index(mapping=str)
+        msa.reassign_index(mapping=lambda e: str(e))
```

Comment:
After offline discussion with @ebolyen, we're going to roll our own! Using `Series.rename` as a passthrough has some other issues, like allowing scalars and list-like to rename the Series' name.